src/execution_tree/primitives/equal_elementwise.cpp

```diff
@@ -0,0 +1,416 @@
+                            "equal::equal2d",
+                            "the operands have incompatible number of "
+                            "dimensions");
+                    }
+                }
+
+                primitive_result_type equal2d1d(
+                    operand_type&& lhs, operand_type&& rhs) const
+                {
+                    std::size_t rhs_size = rhs.dimension(0);
+                    auto lhs_size = lhs.dimensions();
+
+                    if (rhs_size != lhs_size[1])
+                    {
+                        HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                            "equal::equal2d1d",
+                            "the dimensions of the operands do not match");
+                    }
+
+                    // TODO: SIMD functionality should be added, blaze implementation
+                    // is not currently available
+                    for (size_t i = 0UL; i < lhs.matrix().rows(); i++)
+                        blaze::row(lhs.matrix(), i) =
+                            blaze::map(blaze::row(lhs.matrix(), i),
+                                blaze::trans(rhs.vector()),
+                                [](double x, double y) { return x == y; });
+
+                    return primitive_result_type(ir::node_data<bool>{lhs});
+                }
+
+            public:
+                primitive_result_type equal_all(
+                    operand_type&& lhs, operand_type&& rhs) const
+                {
+                    std::size_t lhs_dims = lhs.num_dimensions();
+                    switch (lhs_dims)
+                    {
+                    case 0:
+                        return equal0d(std::move(lhs), std::move(rhs));
+
+                    case 1:
+                        return equal1d(std::move(lhs), std::move(rhs));
+
+                    case 2:
+                        return equal2d(std::move(lhs), std::move(rhs));
+
+                    default:
+                        HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                            "equal::equal_all",
+                            "left hand side operand has unsupported number of "
+                            "dimensions");
+                    }
+                }
+
+            protected:
+                struct visit_equal_elementwise
+                {
+                    template <typename T1, typename T2>
+                    primitive_result_type operator()(T1, T2) const
+                    {
+                        HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                            "equal_elementwise::eval",
+                            "left hand side and right hand side are "
+                            "incompatible "
+                            "and can't be compared");
+                    }
+
+                    primitive_result_type operator()(
+                        ir::node_data<double>&& lhs,
+                        ir::node_data<bool>&& rhs) const
+                    {
```

Comment:
This overload shouldn't be necessary as the overload above (`operator()(T1, T2)`) covers this case.